rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py

```diff
@@ -332,7 +332,16 @@ def test_exec(config):
                             raise TestExecError(
                                 "Command failed....Bucket is not added into reshard queue"
                             )
-                    if config.bucket_sync_crash is True:
+                    if config.bucket_sync_run:
+                        _, source_zone = utils.get_realm_source_zone_info()
+                        log.info(f"Source zone name: {source_zone}")
+                        cmd = f"radosgw-admin bucket sync run --bucket={bucket.name} --source-zone={source_zone}"
+                        out = utils.exec_shell_cmd(cmd)
+                        if out is False:
+                            raise TestExecError(
+                                "Command is throwing error while running bucket sync run"
+                            )
```

Comment:
There are too many `if statements` within this method making it monolithic. IMHO, we can write smaller methods having a specific purpose. For example
```
def check_bucket_syn(config, bucket):
  if not config.bucket_sync_run: return

  log.info("Checking if buckets are syncrhonizing"
  _, source_zone = utils.get_realm_source_zone_info()
  log.debug(f"Source zone name: {source_zone}")
  ...
```